pkg/sink/mysql/db_helper.go

```diff
@@ -235,3 +238,27 @@ func checkTiDBVariable(ctx context.Context, db *sql.DB, variableName, defaultVal
 	// session variable not exists, return "" to ignore it
 	return "", nil
 }
+
+func CheckAndAdjustPassword(ctx context.Context, dbConfig *dmysql.Config, dbConnFactory Factory) (*sql.DB, error) {
+	password := dbConfig.Passwd
+	if dbConnFactory == nil {
+		dbConnFactory = CreateMySQLDBConn
+	}
+	testDB, err := dbConnFactory(ctx, dbConfig.FormatDSN())
+	if err != nil {
+		// If access is denied and password is encoded by base64, try to decoded password.
+		if mysqlErr, ok := errors.Cause(err).(*dmysql.MySQLError); ok && mysqlErr.Number == tmysql.ErrAccessDenied {
+			if dePassword, decodeErr := base64.StdEncoding.DecodeString(password); decodeErr == nil && string(dePassword) != password {
+				log.Info("fizz decode password", zap.String("password", password), zap.String("decode password", string(dePassword)))
```

Comment:
Please don't record the password in the log file

Reply:
done